lib/ansible/parsing/vault/editor.py

```diff
@@ -293,121 +208,205 @@ def rekey_file(self, filename, new_vault_secret, new_vault_id=None):
             f'Re-keyed file "{to_text(filename)}" (decrypted with vault id "{to_text(vault_id_used)}") was encrypted '
             f'with new vault-id "{to_text(new_vault_id)}" and vault secret {to_text(new_vault_secret)}')
 
-    @staticmethod
-    def read_data(filename):
-        try:
-            if filename == '-':
-                data = sys.stdin.buffer.read()
+
+def _shred_file_custom(tmp_path):
```

Comment:
Okay, so your point is, that I should split this PR into multiple smaller ones, so they are easier to review?

Reply:
done

Reply:
It looks like this is still here and a few others:
- _editor_shell_command
- _real_path
- _shred_file
- _shred_file_custom
- confirm
- read_data
- shuffle_files
- write_data

I excluded these from my review. I definitely think we should give code-rewriting the focus in targeted PRs instead of lumping more meaningful changes into this large refactor.

The blanket approach of first just making them all staticmethods, and then standalone functions doesn't feel well-thought out, so if you work on this as a follow-up, I think they should be considered on an individual basis, and be a deliberate code reorganization than just trying to delete the unused `self` any way possible. On the devel branch's version of this function there's a TODO to move it into its own class instead of this, so we should take into account the original intent too.